Double-click-to-edit/DoubleClickToEdit.plugin.js

```diff
@@ -196,10 +244,31 @@ module.exports = class DoubleClickToEdit {
 		if (!message)
 			return;
 
-		if (message.author.id === this.CurrentUserStore.getCurrentUser().id)
-			this.MessageStore.startEditMessage(message.channel_id, message.id, message.content);
-		else if (this.doubleClickToReplySetting)
+		//Check if we're holding down the reply action key
+		let replyKeyHeld = false;
+		switch (this.replyActionKey) {
+			case "shift": replyKeyHeld = e.shiftKey; break;
+			case "ctrl": replyKeyHeld = e.ctrlKey; break;
+			case "alt": replyKeyHeld = e.altKey; break;
+			case "none": replyKeyHeld = !e.altKey && !e.ctrlKey && !e.shiftKey; break;
+			case "any": replyKeyHeld = true; break;
+		}
+
+		//Now we do the same thing with the edit action key
+		let editKeyHeld = false;
+		switch (this.editActionKey) {
+			case "shift": editKeyHeld = e.shiftKey; break;
+			case "ctrl": editKeyHeld = e.ctrlKey; break;
+			case "alt": editKeyHeld = e.altKey; break;
+			case "none": editKeyHeld = !e.altKey && !e.ctrlKey && !e.shiftKey; break;
+			case "any": editKeyHeld = true; break;
+		}
+
+		//If the edit reply key is any, and the reply key is held, we want the reply to be preferred. Otherwise, we want the edit to be preferred.
+		if (replyKeyHeld && (!editKeyHeld || this.editActionKey === "any" && this.replyActionKey !== "any" || message.author.id !== this.CurrentUserStore.getCurrentUser().id))
 			this.replyToMessage(this.getChannel(message.channel_id), message, e);
+		else if (editKeyHeld && message.author.id === this.CurrentUserStore.getCurrentUser().id)
+			this.MessageStore.startEditMessage(message.channel_id, message.id, message.content);
```

Comment:
For the copy method, that would currently not be included I came up with a less explosive and repetative way to check for a pressed key. I made the following class function: 
```
	checkForModifier(enabled, modifier, event) {
		if (enabled)
			switch (modifier) {
				case "shift": return event.shiftKey;
				case "ctrl": return event.ctrlKey;
				case "alt": return event.altKey;
			}
		return false;
	}
```

And to use this for the copy you'd use it like: 
```
		//When selecting text it might be useful to copy.
		const copyKeyHeld = this.checkForModifier(this.doubleClickToCopy, this.copyModifier, e);
		if (copyKeyHeld)
			this.copyToClipboard(document.getSelection().toString());
```
As well as limiting the amount of cases, because it has been replaced by a setting, we can now determine if a modifier is pressed or not in a way that's way more straight forward.

```suggestion
		//Now we do the same thing with the edit and reply modifier
		const editKeyHeld = this.checkForModifier(this.doubleClickToEditModifier, this.editModifier, e),
			replyKeyHeld = this.checkForModifier(this.doubleClickToReplyModifier, this.replyModifier, e);

		//If a modifier is enabled, check if the key is held, otherwise ignore.
		if ((this.doubleClickToEditModifier ? editKeyHeld : true) && message.author.id === this.CurrentUserStore.getCurrentUser().id)
			this.MessageStore.startEditMessage(message.channel_id, message.id, message.content);
		else if ((this.doubleClickToReplyModifier ? replyKeyHeld : true) && this.doubleClickToReply)
			this.replyToMessage(this.getChannel(message.channel_id), message, e);
```

Reply:
All of these make a ton more sense! Lemme implement them, make sure they work as expected, and then make a new commit. However I have an exam tomorrow in uni, so give me a few and I'll get it done ASAP :P

If you would like to do it yourself to make sure it gets done faster, feel free. I should get it done in the next two days (probably sooner), and I'll commit when finished.